src/scripting/lua_ui.cc

```diff
@@ -326,9 +330,427 @@ int LuaPanel::indicate(lua_State* L) {
+             The window will resize itself to accommodate this widget perfectly.
+
+         * ``"box"``: A wrapper around other components with intelligent layouting.
+           It is strongly recommended to layout all your components
+           exclusively with Boxes. Properties:
+
+           * ``"orientation"``: **Mandatory**. The box's layouting direction:
+             ``"vertical"`` or ``"horizontal"``.
+             The shorthands ``"vert"``, ``"v"``, ``"horz"``, and ``"h"`` may be used.
+           * ``"max_x"``: **Optional**. The maximum horizontal size.
+           * ``"max_y"``: **Optional**. The maximum vertical size.
+           * ``"spacing"``: **Optional**. The inner spacing between items.
+           * ``"scrolling"``: **Optional**. Whether the box may scroll if its content is too large.
+
+         * ``"inf_space"``: Only valid as the direct child of a Box. A flexible spacer.
+         * ``"space"``: Only valid as the direct child of a Box. A fixed-size spacer. Property:
+
+           * ``"value"``: **Mandatory**. The size of the space.
+
+         * ``"panel"``: A plain panel that can act as a spacer or as a container for other widgets.
+
+         * ``"textarea"``: A static text area with a single line of text. Properties:
+
+           * ``"text"``: **Mandatory**. The text to display.
+           * ``"font"``: **Mandatory**. The font style to use.
+           * ``"text_align"``: **Optional**. The alignment of the text.
+           * ``"fixed_width"``: **Optional**. If set, the text area's width is fixed instead
+             of resizing to accomodate the text or the parent.
+
+         * ``"button"``: A clickable button. A button must have either a title or an icon,
+           but not both. Properties:
+
+           * ``"title"``: **Optional**. The text on the button.
+           * ``"icon"``: **Optional**. The file path of the button's icon.
+           * ``"style"``: **Optional**. The button's style. One of:
+
+             * ``"primary"``
+             * ``"secondary"``
+             * ``"menu"``
+
+           * ``"visual"``: **Optional**. The button's appearance. One of:
+
+             * ``"raised"``
+             * ``"permpressed"``
+             * ``"flat"``
+
+           * ``"repeating"``: **Optional**. Whether pressing and holding
+             the button generates repeated events.
+
+           * ``"on_click"``: **Optional**. Callback code to run when the
+             button is clicked. To associate actions with a button press,
+             prefer this over the ``on_panel_clicked`` event.
+
+      Note that event callbacks functions must be provided as raw code in string form.
+      During the lifetime of a *toolbar* widget, the Lua Interface used by the game may be reset.
+      Therefore, any callbacks attached to such widgets must not use any functions or variables
+      defined at an arbitrary earlier time by your script -
+      they may have been deleted by the time the callback is invoked.
```

Comment:
An example would be nice. Maybe use the yes/no code with proper callbacks for it?